### ISSUE TITLE
Introduce sub-directory `tasks` for task-sandboxes inside pilot-sandbox

### ISSUE THREAD

**Issue by mtitov** [radical-cybertools/radical.pilot#2482]
Current implementation
```
<remote_workdir>/radical.pilot.sandbox/<session_sandbox>/<pilot_sandbox>/<task_sandbox>
```

Proposal
```
<remote_workdir>/radical.pilot.sandbox/<session_sandbox>/<pilot_sandbox>/tasks/<task_sandbox>
```

@andre-merzky Andre, if this proposal makes sense, should we bring it up within env-isolation development (devel-branch)?

**Comment by andre-merzky** [comment]
I would not mind inverting that: leave the task sandboxes where they are, but move all auxilliary files into a `radical` or `.radical` subtree.  How do you think about that?  Would leave application code unchanged (repex, anything else which makes assumptions relative to the pilot sandbox), and would keep the hierarchy simple (`workdir/resource_sandbox/session_sandbox/pilot_sandbox/task_sandbox`)

**Comment by mtitov** [comment]
@andre-merzky that will require more effort, but in this case we can review all data types within `pilot_sandbox` (just relocate them all together into sub-dir doesn't sound right)
- user staged data - should stay within `pilot://` space;
- auxiliary scripts/tools, e.g., `gtod`, `prof` - could be moved into sub-dir `utils` or stay in `pilot_sandbox` (or even being moved into the upper layer, into `session_sandbox`);
- logs, profiles, err-/out-files - could be moved into one sub-dir `register` (?) or separate ones per type;
- configs and json-files - could be moved into sub-dir `configs`;

eh, having tasks grouped inside a dedicated directory still sounds good to me, but understand having certain hierarchy 

p.s. should [RFC.6](https://github.com/radical-cybertools/radical.pilot/wiki/RFC.6) (Relative data staging) be part of this ticket?

**Comment by andre-merzky** [comment]
I agree with subdirs for different types of data. I would probably propose to sort them a bit differently, but in the same gist.  Like: `env/` `profiles/`, `logfiles/`, etc., but leave smaller ones (`gtod`, `prof`, `bootstrap_0.sh`, agent configs ...) directly in that  dir.  FWIW, most of the config and json files hopefully migrate to the registry over time.  But I don't think that's very different from what you said...

Moving under `.radical` or `radical` would avoid clashes with task sandbox names and staged data (users can define those after all).

> p.s. should RFC.6 (Relative data staging) be part of this ticket?

I think that RFC is in sync with this proposal already, isn't it?

**Comment by mtitov** [comment]
As a summary - focus on grouping logs and profiles, and keep sandbox-hierarchy as it is (configuration files planned to be moved into registry, thus here they are out of scope)

**Comment by mtitov** [comment]
Closed in favor of #2486
